src/zenml/materializers/built_in_materializer.py

```diff
@@ -294,20 +294,40 @@ def load(self, data_type: Type[Any]) -> Any:
         else:
             metadata = yaml_utils.read_json(self.metadata_path)
             outputs = []
-            for path_, type_str in zip(metadata["paths"], metadata["types"]):
-                type_ = find_type_by_str(type_str)
-                materializer_class = default_materializer_registry[type_]
-                materializer = materializer_class(uri=path_)
-                element = materializer.load(type_)
-                outputs.append(element)
+
+            # Backwards compatibility for zenml <= 0.37.0
+            if isinstance(metadata, dict):
+                for path_, type_str in zip(
+                    metadata["paths"], metadata["types"]
+                ):
+                    type_ = find_type_by_str(type_str)
+                    materializer_class = default_materializer_registry[type_]
+                    materializer = materializer_class(uri=path_)
+                    element = materializer.load(type_)
+                    outputs.append(element)
+
+            # New format for zenml >= 0.37.1
```

Comment:
```suggestion
            # New format for zenml > 0.37.0
```

Who knows, it might be 0.38.0 😉